pkg/controller/infinispan/resources/operatorconfig/operatorconfig_controller.go

```diff
@@ -0,0 +1,137 @@
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	ControllerName = "operator-config-controller"
+	configMapName  = "infinispan-operator-config"
+)
+
+const (
+	grafanaDashboardNamespaceKey  = "grafana.dashboard.namespace"
+	grafanaDashboardNameKey       = "grafana.dashboard.name"
+	grafanaDashboardMonitoringKey = "grafana.dashboard.monitoring.key"
+)
+
+var ctx = context.Background()
+
+// DefaultConfig default configuration for operator config controller
+var defaultOperatorConfig = map[string]string{
+	grafanaDashboardMonitoringKey: "middleware",
+	grafanaDashboardNameKey:       "infinispan",
+}
+
+// ReconcileInfinispan reconciles a Infinispan object
+type ReconcileOperatorConfig struct {
+	client client.Client
+	scheme *runtime.Scheme
+	Log    logr.Logger
+	Kube   *kube.Kubernetes
+}
+
+func Add(mgr manager.Manager) error {
+	return addOperatorConfig(mgr)
+}
+
+// addOperatorConfig installs a reconciler for a specific object.
+// Only events coming from operatorNs/infinispan-operator-config will be processed
+func addOperatorConfig(mgr manager.Manager) error {
+	r := &ReconcileOperatorConfig{client: mgr.GetClient(), scheme: mgr.GetScheme(), Log: logf.Log.WithName("operator-config-controller"), Kube: kube.NewKubernetesFromController(mgr)}
+	// Create a new controller
+	c, err := controller.New(ControllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+	operatorNS, err := getOperatorNamespace()
+	if err != nil {
+		return err
+	}
+	operatorConfigPredicate := predicate.Funcs{
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return e.Meta.GetName() == configMapName && e.Meta.GetNamespace() == operatorNS
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return e.Meta.GetName() == configMapName && e.Meta.GetNamespace() == operatorNS
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return e.MetaNew.GetName() == configMapName && e.MetaNew.GetNamespace() == operatorNS
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return e.Meta.GetName() == configMapName && e.Meta.GetNamespace() == operatorNS
+		},
+	}
+	// Watch for changes to global operator config
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, operatorConfigPredicate)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileOperatorConfig{}
+var currentConfig map[string]string = make(map[string]string)
```

Comment:
It would be nice to have this config available to other controllers, but due to concurrent access we need to use a blocking mechanism for it

Reply:
yeah but I would do that in a separate PR when concurrent access will be needed